### PR TITLE
Godwken benchmark

### PR DESCRIPTION
**Pull request by magicalne**
This pr contains such changes:

- Turns gw-tools to lib to share common code.
- Add gw-benchmark as a new crate bin.

One thing to mention:
Since the original godwoken_rpc is a blocking client and **blocking** reqwest client shouldn't be used in the context of "async". A blocking client can easily get into a deadlock in a tokio app. So I added an async version of the godwoken RPC client in the benchmark crate.

### CODE REVIEW

File: crates/tools/src/godwoken_rpc.rs [hunk 1]
```diff
@@ -72,7 +72,7 @@ impl GodwokenRpcClient {
         short_address: JsonBytes,
     ) -> Result<Option<H256>> {
         let params = serde_json::to_value((short_address,))?;
-
+        dbg!("get_script_hash_by_short_address");
```
Comment (jjyr) [inline crates/tools/src/godwoken_rpc.rs:75]: Remove dbg before send PR

**Review by jjyr — CHANGES_REQUESTED** [review]

**Review by jjyr — DISMISSED** [review]

**Comment by jjyr** [comment]
Is this PR ready for merge?

File: crates/benchmark/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    fn default() -> Self {
+        Self {
+            interval: 1000,
+            batch: 10,
+            timeout: 120,
+            account_path: PathBuf::from("./accounts"),
+            gw_rpc_url: String::from("http://localhost:8119"),
+            polyman_url: String::from("http://localhost:6102"),
+            scripts_deploy_path: PathBuf::from("./scripts_deploy_results.json"),
+            rollup_type_hash: String::from("0x"),
```
Comment (Flouse) [inline crates/benchmark/src/config.rs:27]: Next version, maybe we could read `rollup_type_hash` from `gw-config.toml` and remove this config from here.

**Review by Flouse — DISMISSED** [review]

File: crates/benchmark/src/config.rs (hunk 2 above)
Comment (magicalne) [reply crates/benchmark/src/config.rs:27]: Let's just pass the parameters that we actually need. If we could use more configs from `config.toml`, then I'd like to involve `config.toml` instead.

**Comment by jjyr** [comment]
Please cherry-pick this PR to develop branch.

**Comment by jjyr** [comment]
Please cherry-pick this PR to develop branch and open a new PR

**Pull request closed without merging**